src/data_provider/src/packages/rpmPackageManager.cpp

```diff
@@ -0,0 +1,157 @@
+#include "rpmPackageManager.h"
+
+#include <exception>
+#include <map>
+#include <stdexcept>
+#include <vector>
+#include <iostream>
+
+// For O_RDONLY
+#include <fcntl.h>
+
+bool instantiated = false;
+
+RpmPackageManager::RpmPackageManager(std::shared_ptr<IRpmLibWrapper>&& wrapper)
+    : m_rpmlib{wrapper}
+{
+    if (instantiated)
+    {
+        throw std::runtime_error("there is another RPM instance already created");
+    }
+
+    if (m_rpmlib->rpmReadConfigFiles(nullptr, nullptr))
+    {
+        throw std::runtime_error("rpmReadConfigFiles failed");
+    }
+
+    instantiated = true;
+}
+
+RpmPackageManager::~RpmPackageManager()
+{
+    m_rpmlib->rpmFreeRpmrc();
+    instantiated = false;
+}
+
+std::string RpmPackageManager::Iterator::getAttribute(rpmTag tag)
+{
+    std::string str;
```

Comment:
Maybe `attribute` or `retval` is more descriptive

Reply:
Yeah – this is definitely a style thing. This is favored in other projects/languages but I've seen `retval` being used a lot around here so it makes sense to stick to that. :)